docs/en/Community-Articles/2024-05-07-using-fluent-validation-with-abp/post.md

```diff
@@ -0,0 +1,95 @@
+
+## How to Use Fluent Validation in ABP?
+![fluent validation](./images/abp-fluentvalidation.png)
+
+Using Fluent validation with Abp is quite simple. Open a command line window in the folder of the project (.csproj file) and type the following command: 
+abp add-package Volo.Abp.FluentValidation
+If you have not created your abp project, review the [steps to create](https://docs.abp.io/en/abp/latest/Tutorials/Todo/Overall) it now.
+
+Create your product object like this: 
+````csharp
+  public class Product: FullAuditedAggregateRoot<Guid>
+   {
+       public string Name { get; set; } = string.Empty;
+       public decimal Price { get; set; }
+       public int Stock { get; set; }
+       public string? LicenseKey { get; set; }
+    }
+````
+Create the controller of your product object by following the [documentation](https://docs.abp.io/en/abp/latest/Tutorials/Todo/Index?UI=NG&DB=EF) 
+You must have ProductCreateDto :
+````csharp
+      public class ProductCreateDto
+    {
+        public string Name { get; set; } 
+        public decimal Price { get; set; }
+        public int Stock { get; set; }
+        public string? LicenseKey { get; set; }
+     }
+````
+Create the ProductCreateDtoValidator class in the products folder in the Application.Contacts layer. 
+````csharp
+  public class ProductCreateDtoValidator :AbstractValidator<ProductCreateDto>
+{
+    public ProductCreateDtoValidator()
+    {
+        RuleFor(p=>p.Name).
+          NotEmpty().
+            WithMessage("Product name cannot be empty");
+        
+        RuleFor(p=>p.Name).
+            MinimumLength(3).
+            MaximumLength(100).
+            WithMessage("Product name must be between 3 and 100 characters");
+
+        RuleFor(p => p.Stock).
+            GreaterThanOrEqualTo(0).
+            WithMessage("Product stock should not be negative");
+
+        RuleFor(p => p.Price).
+           GreaterThanOrEqualTo(0).
+           WithMessage("Product Price should not be negative");
+    }
+}
+````
+Fluent validation provides you with ready-made methods that you can use to write validations very easily. 
+You can add a method to write your own customized validation code. For example, let's write the code that requires the license key field to contain the word “key”.
+````csharp
+  private bool ContainsKey(string arg)
+  {
+      return arg.IndexOf("key", StringComparison.OrdinalIgnoreCase) >= 0;
+  }
+````
+
+Add the code to the constructor to use this method: 
+
+````csharp
+RuleFor(p => p.LicenseKey).
+	Must(ContainsKey).
+	WithMessage("Product license key must contain the word 'key'");
+	
+````
+
+
+When I write code that does not follow the rule, we get this error	
```

Comment:
Before showing what will happen if the validation fails, I think you should show the application service implementation and how to use it, and then, write this message and show what will happen if validation fails. In this way, the reader will be able to directly follow the article and try the library. 

Because, in this current way, it's not clear how to get the benefit of these validation rules.

---

Also, please update this line as below:

```suggestion
If one of the validation rules does not meet the rules, then the following error will be received for the custom rule that we defined:
```

Reply:
I have attached an article explaining how to get the error:

Try to add data that does not meet the validation rules

![Adding Incorrect Data](./images/adding-incorrect-data.png)